src/main/java/com/redhat/cloud/policies/app/model/history/PoliciesHistoryRepository.java

```diff
@@ -0,0 +1,162 @@
+package com.redhat.cloud.policies.app.model.history;
+
+import com.redhat.cloud.policies.app.model.filter.Filter;
+import com.redhat.cloud.policies.app.model.pager.Pager;
+import io.quarkus.panache.common.Sort;
+import org.hibernate.Session;
+import org.jboss.logging.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.persistence.Query;
+import javax.persistence.TypedQuery;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import static com.redhat.cloud.policies.app.model.filter.Filter.Operator.LIKE;
+
+@ApplicationScoped
+public class PoliciesHistoryRepository {
+
+    private static final Logger LOGGER = Logger.getLogger(PoliciesHistoryRepository.class);
+
+    @Inject
+    Session session;
```

Comment:
The rest of the application uses `Panache`. Is this more effective or better in any way? 
I understand this can be more flexible, but do we do something that can't be done in `Panache` ? A `Session` for custom queries can still be used IIUC.

Reply:
Some of the code could probably be simplified with stuff like:
```
  PanacheQuery<Foo> panacheQuery = find(
            filter.getQuery(),
            pager.getSort(),
            filter.getParameters()
    );
```

and 
```
return new Page<>(
            panacheQuery.list(),
            pager,
            panacheQuery.count()
    );
```

Reply:
I tried with Panache first, spent a lot of time on it for no good reason and then moved on to something easier to implement (at least for me). 😃 

They say `Panache focuses on making your entities trivial and fun to write in Quarkus` but I'm still looking for the fun part after 2 years knowing Panache. I guess it's great for simple cases when you don't need to build a dynamic query.

Anyway, I don't think it's a problem to have a mix between Panache and plain old HQL/SQL in the same app. It all depends on Hibernate ORM in the end.

Reply:
If you really prefer Panache, I can give it another try.

Reply:
It's fine, maybe we can find the fun in a separate PR.